src/magnify/apps/core/models.py

```diff
@@ -8,7 +8,48 @@
 
 
 class User(auth_models.AbstractUser):
-    """User model which follow courses or manage backend (is_staff)"""
+    """User model"""
+
+    username_validator = RegexValidator(
+        "^[a-zA-Z0-9_-]{3,16}$",
+        message="""Username must be between 3 and 16 characters long
+            and contain only letters, numbers, underscores and hyphens.""",
+    )
+
+    username = models.CharField(
+        _("username"),
+        max_length=150,
+        unique=True,
+        validators=[username_validator],
```

Comment:
I'm not fan of having validation on the model. In an API I prefer having them on the serializer unless you plan to use this model in a management command.

Reply:
why are you against model validation for this case? Shouldn't it even be a database constraint if we really want to be sure our users are properly named?

Reply:
Most of time validators on the database avoid future modification. Once modified, part of your data will become invalid and you will not be able to modify the all model without doing a sql migration.

Reply:
I would consider this a feature but we can discuss irl.

Reply:
I really don't know how to add validators in the serializer. If the feature is seen as necessary, some help would be welcome ;)